adam/curriculum/verbs_with_dynamic_prepositions_curriculum.py

```diff
@@ -883,6 +887,210 @@ def _fall_in_front_of_behind_template(
     )
 
 
+# PUT templates
+
+
+def _put_on_template(
+    agent: TemplateObjectVariable,
+    theme: TemplateObjectVariable,
+    goal_reference: TemplateObjectVariable,
+    background: Iterable[TemplateObjectVariable],
+) -> Phase1SituationTemplate:
+    return Phase1SituationTemplate(
+        f"{agent.handle}-puts-{theme.handle}-on-{goal_reference.handle}",
+        salient_object_variables=[agent, theme, goal_reference],
+        background_object_variables=background,
+        actions=[
+            Action(
+                PUT,
+                argument_roles_to_fillers=[
+                    (AGENT, agent),
+                    (THEME, theme),
+                    (
+                        GOAL,
+                        Region(
+                            goal_reference,
+                            distance=EXTERIOR_BUT_IN_CONTACT,
+                            direction=GRAVITATIONAL_UP,
+                        ),
+                    ),
+                ],
+            )
+        ],
+        constraining_relations=flatten_relations(
+            [bigger_than(agent, theme), bigger_than(goal_reference, theme)]
```

Comment:
nit: this can be written as `bigger_than([agent, goal_reference], theme)` as the `bigger_than` can take an iterable input. I'm going to mark all instances of this with 📓 for easy reference